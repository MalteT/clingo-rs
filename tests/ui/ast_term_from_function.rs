use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let term2 = ast::Term::from(sym);
    let args = vec![term1,term2];
    let mut fun = ast::Function::new("name", &args).unwrap();
    let term3 = ast::Term::from(&fun);

    let mut fun2 = ast::Function::new("name2", &args).unwrap();
    let term4 = ast::Term::external_function(&fun2);

    fun = ast::Function::new("name", &args).unwrap();
    fun2 = ast::Function::new("name", & args).unwrap();
    drop(args);
    let _end = (term3, term4, fun, fun2);
}