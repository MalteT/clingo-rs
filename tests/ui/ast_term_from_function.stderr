error[E0506]: cannot assign to `fun` because it is borrowed
  --> $DIR/ast_term_from_function.rs:14:5
   |
9  |     let term3 = ast::Term::from(&fun);
   |                                 ---- borrow of `fun` occurs here
...
14 |     fun = ast::Function::new("name", &args).unwrap();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `fun` occurs here
...
17 |     let _end = (term3, term4, fun, fun2);
   |                 ----- borrow later used here

error[E0506]: cannot assign to `fun2` because it is borrowed
  --> $DIR/ast_term_from_function.rs:15:5
   |
12 |     let term4 = ast::Term::external_function(&fun2);
   |                                              ----- borrow of `fun2` occurs here
...
15 |     fun2 = ast::Function::new("name", & args).unwrap();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `fun2` occurs here
16 |     drop(args);
17 |     let _end = (term3, term4, fun, fun2);
   |                        ----- borrow later used here

error[E0505]: cannot move out of `args` because it is borrowed
  --> $DIR/ast_term_from_function.rs:16:10
   |
8  |     let mut fun = ast::Function::new("name", &args).unwrap();
   |                                              ----- borrow of `args` occurs here
...
16 |     drop(args);
   |          ^^^^ move out of `args` occurs here
17 |     let _end = (term3, term4, fun, fun2);
   |                               --- borrow later used here
